snuba/query/logical.py

```diff
@@ -254,6 +254,14 @@ def get_prewhere_ast(self) -> Optional[Expression]:
     def set_prewhere_ast_condition(self, condition: Optional[Expression]) -> None:
         self.__prewhere = condition
 
+    def add_prewhere_condition_to_ast(self, condition: Expression) -> None:
+        if not self.__prewhere:
+            self.__prewhere = condition
+        else:
+            self.__prewhere = binary_condition(
+                None, BooleanFunctions.AND, condition, self.__prewhere
+            )
+
```

Comment:
You should not need this. See my comment on the prewhere in the processor class.